src/lib/ptree/json.lua

```diff
@@ -83,34 +98,87 @@ local function read_json_string(input)
       end
       table.insert(parts, part)
    end
+   skip_whitespace(input)
```

Comment:
Can we move this to *before* parsing an object?  Otherwise we run the risk of blocking, when we actually have an object to return!

Reply:
Hmm - I understand what you're getting at and it makes sense. `read_json` already skips whitespace at the start of parsing a type, so shouldn't be necessary to add any further `skip_whitespace` back in, but removing this specific instance causes the tests to fail as the 'sentinel' whitespace in the test is read on line 259 (`assert(read_json(tmp) == nil)`) and triggers an error in `read_scalar` as that is the default parse method.

I think moving `read_whitespace` in `read_json` above the EOF check will fix this (doing this makes the tests pass at least), as if there's any whitespace left over (either trailing or leading) it'll be dropped, and then the EOF check will return `nil`. 

I can't see any issue with this as I *think* that `take_while` will not consume anything if the stream is EOF. I'll push the change as described above.

Thoughts?